### PR TITLE
Fix `<Appbar>` buttons don't have consistent spacing

### LINKED ISSUE #8347: IconButton size on AppBar is not very consistent
**What you were expecting:**

I expect all the IconButton on AppBar to have consistent size. (or a feature to allow better override of default size)

**What happened instead:**

Some of IconButtons are large and some are medium, also the spacing, they are not very consistent.

**Steps to reproduce:**

**Related code:**

Medium by default https://github.com/marmelab/react-admin/blob/a408de0db2c6db0281b3d39e27c1f2de3deed44e/packages/ra-ui-materialui/src/button/ToggleThemeButton.tsx#L39

Medium by default https://github.com/marmelab/react-admin/blob/64a9091c49a2a36a8974f1c261727edc64eeb39e/packages/ra-ui-materialui/src/preferences/InspectorButton.tsx#L40

large https://github.com/marmelab/react-admin/blob/a408de0db2c6db0281b3d39e27c1f2de3deed44e/packages/ra-ui-materialui/src/button/RefreshIconButton.tsx#L32

large https://github.com/marmelab/react-admin/blob/a408de0db2c6db0281b3d39e27c1f2de3deed44e/packages/ra-ui-materialui/src/layout/UserMenu.tsx#L107

large https://github.com/marmelab/react-admin/blob/479bc4af86d8a6fa31022ab57a7fe72428c82e36/packages/ra-ui-materialui/src/layout/SidebarToggleButton.tsx#L32

**Other information:**

**Environment**

* React-admin version: 4.5.0
* Last version that did not exhibit the issue (if applicable): N/A
* React version: 18.2.0
* Browser:
* Stack trace (in case of a JS error):

### PR DESCRIPTION
**Pull request by fzaninotto**
## Before

image

## After

image

Closes #8347

### CODE REVIEW

File: packages/ra-ui-materialui/src/button/LocalesMenuButton.tsx [hunk 1]
```diff
@@ -86,18 +84,12 @@ export const LocalesMenuButton = (props: LocalesMenuButtonProps) => {
...
-export const LocalesMenuButtonClasses = {
-    selectedLanguage: `${PREFIX}-selectedLanguage`,
-};
+export const LocalesMenuButtonClasses = {};
 
 const Root = styled(Box, {
     name: PREFIX,
     overridesResolver: (props, styles) => styles.root,
-})(({ theme }) => ({
-    [`& .${LocalesMenuButtonClasses.selectedLanguage}`]: {
```
Comment (slax57) [inline packages/ra-ui-materialui/src/button/LocalesMenuButton.tsx:97]: This change requires a doc update

File: packages/ra-ui-materialui/src/layout/SidebarToggleButton.tsx [hunk 2]
```diff
@@ -29,11 +29,7 @@ export const SidebarToggleButton = (props: SidebarToggleButtonProps) => {
             )}
             enterDelay={500}
         >
-            <StyledIconButton
-                color="inherit"
-                onClick={() => setOpen(!open)}
-                size="large"
-            >
+            <StyledIconButton color="inherit" onClick={() => setOpen(!open)}>
```
Comment (slax57) [inline packages/ra-ui-materialui/src/layout/SidebarToggleButton.tsx:32]: #8350 also suggested to pass the `...rest` props to the button, don't you think we should do it too?

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/layout/SidebarToggleButton.tsx (hunk 2 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/layout/SidebarToggleButton.tsx:32]: > #8350 also suggested to pass the `...rest` props to the button, don't you think we should do it too?

+1 on this

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/layout/SidebarToggleButton.tsx (hunk 2 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/layout/SidebarToggleButton.tsx:32]: I don't see the need for this

**Review by WiXSL — APPROVED** [review]

**Pull request merged**